# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: intl/hyphenation/glue/nsHyphenator.cpp
# Commit: 4143daaac691
# Full Hash: 4143daaac691ea2ce9997c4e41e54798d40a439d
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2020-10-29 15:52:12
# Description:
#   Bug 1672388 - Validate hyphenation data before freezing the shared memory, which will unmap it. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D94320
# ==============================================================================

diff -r d7a70f3f59e7 -r 4143daaac691 intl/hyphenation/glue/nsHyphenator.cpp
--- a/intl/hyphenation/glue/nsHyphenator.cpp	Thu Oct 29 10:14:37 2020 +0000
+++ b/intl/hyphenation/glue/nsHyphenator.cpp	Thu Oct 29 12:27:49 2020 +0000
@@ -161,12 +161,13 @@
         bytesRead != available) {
       return nullptr;
     }
-    if (!shm->Freeze()) {
+
+    if (!mapped_hyph_is_valid_hyphenator(
+            reinterpret_cast<const uint8_t*>(buffer), bytesRead)) {
       return nullptr;
     }
 
-    if (!mapped_hyph_is_valid_hyphenator(
-            reinterpret_cast<const uint8_t*>(buffer), bytesRead)) {
+    if (!shm->Freeze()) {
       return nullptr;
     }
 
